### PR TITLE
xorg-server: add elogind build_option

### PR DESCRIPTION
**Pull request by maxice8**

### CODE REVIEW

File: srcpkgs/xorg-server/template [hunk 1]
```diff
@@ -29,6 +30,9 @@ replaces="xf86-video-modesetting>=0"
 subpackages="xorg-server-devel xorg-server-xdmx xorg-server-xephyr xorg-server-xnest xorg-server-xvfb xorg-server"
 conf_files="/etc/X11/Xwrapper.config"
 
+build_options="elogind"
+desc_option_elogind="Rootless Xorg support with elogind"
```
Comment (CameronNemo) [inline srcpkgs/xorg-server/template]: do we want to add this description in `common`

**Comment by maxice8** [comment]
On Sat, Nov 17, 2018 at 12:38:15PM -0800, Cameron Nemo wrote:
> CameronNemo commented on this pull request.
> 
> 
> 
> > @@ -29,6 +30,9 @@ replaces="xf86-video-modesetting>=0"
>  subpackages="xorg-server-devel xorg-server-xdmx xorg-server-xephyr xorg-server-xnest xorg-server-xvfb xorg-server"
>  conf_files="/etc/X11/Xwrapper.config"
>  
> +build_options="elogind"
> +desc_option_elogind="Rootless Xorg support with elogind"
> 
> do we want to add this description in `common`
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/4843#pullrequestreview-176066981

No, it is unique to xorg-server and not widely used.

Elogind provides more than just rootless <program> options.

**Comment by maxice8** [comment]
common is for when we have lots of build options for the same thing like generating Vala bindings or gobject introspection data

**Comment by Gottox** [comment]
I would either use a more generic description in common or just leave it there.

**Comment by maxice8** [comment]
On Sun, Nov 18, 2018 at 11:31:23AM -0800, Enno Boland wrote:
> I would either use a more generic description in common or just leave it there.
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/4843#issuecomment-439718934

I'd rather leave it here, elogind is used in lots of places in different ways while vala and gir are pretty much "generate this" and other stuff in common/ is "enable support for X"

it is also nice to have it describe what exactly it does

File: srcpkgs/xorg-server/files/Xwrapper.config [hunk 2]
```diff
@@ -1 +1 @@
-needs_root_rights = yes
+needs_root_rights = auto
```
Comment (Gottox) [inline srcpkgs/xorg-server/files/Xwrapper.config]: I'm not sure if we should do this now.

**Comment by prspkt** [comment]
Setting `needs_root_rights = yes` works fine on x86_64-musl with the modesetting driver.

**Comment by maxice8** [comment]
@prspkt needs_root_rights is the default on Void and will always work in those cases, the point here is to avoid having to need root rights by using elogind.

**Comment by prspkt** [comment]
@maxice8 sorry, my mistake :( i actually meant `needs_root_rights = no`.

File: srcpkgs/xorg-server/files/Xwrapper.config (hunk 2 above)
Comment (maxice8) [reply srcpkgs/xorg-server/files/Xwrapper.config]: reverted

**Pull request merged**